lib/fizzy/parser.cpp

```diff
@@ -559,7 +571,7 @@ Module parse(bytes_view input)
     }
 
     if (!module.datasec.empty() && !module.has_memory())
-        throw validation_error{"data section encountered without a memory section"};
```

Comment:
These error messages were more descriptive and user-friendly, though, not sure if it makes sense to always follow the reference interpreter's messages.

Reply:
The `unexpected tableidx value 1` is not better.